diff --git a/dist/pinia.cjs b/dist/pinia.cjs
index 7361d77344eff4e15aa9446d76e2e671962eee92..024b74c30506cadaa7bd8b3deef6ae619b251be6 100644
--- a/dist/pinia.cjs
+++ b/dist/pinia.cjs
@@ -1211,7 +1211,7 @@ function skipHydrate(obj) {
  * @returns true if `obj` should be hydrated
  */
 function shouldHydrate(obj) {
-    return !isPlainObject(obj) || !obj.hasOwnProperty(skipHydrateSymbol);
+    return !isPlainObject(obj) || !Object.prototype.hasOwnProperty.call(obj, skipHydrateSymbol);
 }
 const { assign } = Object;
 function isComputed(o) {
diff --git a/dist/pinia.esm-browser.js b/dist/pinia.esm-browser.js
index 2a7315681dcb2510997941ed7e79d656a4522254..a9dbeee4e125a4975fa77da2d2ae51ddd69cdcbc 100644
--- a/dist/pinia.esm-browser.js
+++ b/dist/pinia.esm-browser.js
@@ -1204,7 +1204,7 @@ function skipHydrate(obj) {
  * @returns true if `obj` should be hydrated
  */
 function shouldHydrate(obj) {
-    return !isPlainObject(obj) || !obj.hasOwnProperty(skipHydrateSymbol);
+    return !isPlainObject(obj) || !Object.prototype.hasOwnProperty.call(obj, skipHydrateSymbol);
 }
 const { assign } = Object;
 function isComputed(o) {
diff --git a/dist/pinia.iife.js b/dist/pinia.iife.js
index 82a013975fdbe48429ed8a953c2a09e475a4413f..f894fc0dc0135a41326ed2e5c0f3523c721ae55c 100644
--- a/dist/pinia.iife.js
+++ b/dist/pinia.iife.js
@@ -1372,7 +1372,7 @@ var Pinia = (function (exports, vueDemi) {
    * @returns true if `obj` should be hydrated
    */
   function shouldHydrate(obj) {
-      return !isPlainObject(obj) || !obj.hasOwnProperty(skipHydrateSymbol);
+      return !isPlainObject(obj) || !Object.prototype.hasOwnProperty.call(obj, skipHydrateSymbol);
   }
   const { assign } = Object;
   function isComputed(o) {
diff --git a/dist/pinia.mjs b/dist/pinia.mjs
index 0092356ccb9046da25e20360292254ea12f8c16a..e52dfe3b790314b3a321e8eb045e30be59539707 100644
--- a/dist/pinia.mjs
+++ b/dist/pinia.mjs
@@ -1209,7 +1209,7 @@ function skipHydrate(obj) {
  * @returns true if `obj` should be hydrated
  */
 function shouldHydrate(obj) {
-    return !isPlainObject(obj) || !obj.hasOwnProperty(skipHydrateSymbol);
+    return !isPlainObject(obj) || !Object.prototype.hasOwnProperty.call(obj, skipHydrateSymbol);
 }
 const { assign } = Object;
 function isComputed(o) {
diff --git a/dist/pinia.prod.cjs b/dist/pinia.prod.cjs
index 959c6611601956bb17dbd78a39b57d6c333817ab..ae344f93a89d0712dd505b2c4e3b4f2d8f71eccf 100644
--- a/dist/pinia.prod.cjs
+++ b/dist/pinia.prod.cjs
@@ -228,7 +228,7 @@ function skipHydrate(obj) {
  * @returns true if `obj` should be hydrated
  */
 function shouldHydrate(obj) {
-    return !isPlainObject(obj) || !obj.hasOwnProperty(skipHydrateSymbol);
+    return !isPlainObject(obj) || !Object.prototype.hasOwnProperty.call(obj, skipHydrateSymbol);
 }
 const { assign } = Object;
 function isComputed(o) {
